tests/functions/test_pivot_longer.py

```diff
@@ -76,6 +76,472 @@ def test_df():
+def test_presence_index(df, index):
+    "Raise ValueError if labels in `index` do not exist."
+    with pytest.raises(ValueError):
+        df.pivot_longer(index=index)
+
+
+@pytest.mark.parametrize("df,column", column_presence_checks)
+def test_presence_columns(df, column):
+    "Raise ValueError if labels in `column_names` do not exist."
+    with pytest.raises(ValueError):
+        df.pivot_longer(column_names=column)
+
+
+@pytest.mark.parametrize("df,names_to, names_sep", names_sep_not_required)
+def test_name_sep_names_to_len(df, names_to, names_sep):
+    """
+    Raise ValueError if the `names_to` is a string, or `names_to` is a
+    list/tuple, its length is one, and `names_sep` is provided.
+    """
+    with pytest.raises(ValueError):
+        df.pivot_longer(names_to=names_to, names_sep=names_sep)
+
+
+@pytest.mark.parametrize("df,names_to, names_sep", names_sep_type_check)
+def test_name_sep_wrong_type(df, names_to, names_sep):
+    "Raise TypeError if the wrong type is provided for `names_sep`."
+    with pytest.raises(TypeError):
+        df.pivot_longer(names_to=names_to, names_sep=names_sep)
+
+
+@pytest.mark.parametrize(
+    "df,names_to, names_pattern", names_pattern_type_check
+)
+def test_name_pattern_wrong_type(df, names_to, names_pattern):
+    "Raise TypeError if the wrong type provided for `names_pattern`."
+    with pytest.raises(TypeError):
+        df.pivot_longer(names_to=names_to, names_pattern=names_pattern)
+
+
+def test_names_pattern_wrong_subtype():
+    """
+    Raise TypeError if `names_pattern` is a list/tuple
+    and wrong subtype is supplied.
+    """
+    with pytest.raises(TypeError):
+        df_checks.pivot_longer(
+            names_to=["variable", "value"], names_pattern=[1, "rar"]
+        )
+
+
+def test_names_pattern_names_to_wrong_type():
+    """
+    Raise TypeError if `names_pattern` is a list/tuple
+    and wrong type is supplied for `names_to`.
+    """
+    with pytest.raises(TypeError):
+        df_checks.pivot_longer(
+            names_to={"variable, value"}, names_pattern=["1", "rar"]
+        )
+
+
+def test_column_level_wrong_type(df_multi):
+    "Raise TypeError if wrong type is provided for `column_level`."
+    with pytest.raises(TypeError):
+        df_multi.pivot_longer(index="name", column_level={0})
+
+
+def test_sort_by_appearance(test_df):
+    "Raise error if `sort_by_appearance` is not boolean."
+    with pytest.raises(TypeError):
+        test_df.pivot_longer(
+            names_to=[".value", "value"],
+            names_sep="_",
+            sort_by_appearance="TRUE",
+        )
+
+
+def test_ignore_index(test_df):
+    "Raise error if `ignore_index` is not boolean."
+    with pytest.raises(TypeError):
+        test_df.pivot_longer(
+            names_to=[".value", "value"], names_sep="_", ignore_index="TRUE"
+        )
+
+
+def test_duplicate_dot_value(test_df):
+    "Raise error if `names_to` contains more than 1 `.value."
```

Comment:
docstring.

Also, do you mean: "Raise error if `names_to` contains more than one value"?